lib/collection/src/collection_manager/segments_updater.rs

```diff
@@ -125,6 +127,8 @@ pub(crate) fn delete_vectors(
                 }
                 Ok(res)
             },
+            // Apply delete to all point versions
+            true,
```

Comment:
I think this should be `false`. Points with different versions might have different payloads, deleting vectors will create unresolvable conflict on merge

Reply:
Do you mean in case a payload filter is used?

I'm not too worried about it as we should still only use the latest point version for retrieval.

But since this is not strictly related to point deletion, and since this on its own cannot revive old keys, I'm happy with changing this one to false 👍

Reply:
f492b6c3f86e309134dda2f8ebe43eecd2c16a02